provider/rancher/rancher.go

```diff
@@ -393,11 +392,8 @@ func parseRancherData(environments []*rancher.Environment, services []*rancher.S
 			}
 
 			for _, container := range containers {
-				for key, value := range container.Labels {
-
-					if key == "io.rancher.stack_service.name" && value == rancherData.Name {
-						rancherData.Containers = append(rancherData.Containers, container.PrimaryIpAddress)
-					}
+				if containerFilter(container) && container.Labels["io.rancher.stack_service.name"] == rancherData.Name {
```

Comment:
It would be even better to write `if container.Labels["io.rancher.stack_service.name"] == rancherData.Name && containerFilter(container) {`, but meh :)